src/Sulu/Bundle/AdminBundle/Resources/js/stores/userStore/userStore.js

```diff
@@ -60,13 +61,26 @@ class UserStore {
             this.persistentSettings.set(key, persistentSettings[key]);
         });
 
-        // TODO this code should be adjusted/removed when a proper content-locale handling is implemented
-        // load and use first (default) localization of first webspace as content-locale for the user
-        localizationStore.loadLocalizations().then(action((localizations) => {
-            const defaultLocalizations = localizations.filter((localization) => localization.default);
-            const fallbackLocalization = defaultLocalizations.length ? defaultLocalizations[0] : localizations[0];
-            this.contentLocale = fallbackLocalization ? fallbackLocalization.locale : this.contentLocale;
-        }));
+        const contentLocale = this.persistentSettings.get(CONTENT_LOCALE_SETTING_KEY);
+
+        if (contentLocale) {
+            this.contentLocale = contentLocale;
+        }
+
+        if (!this.contentLocale) {
+            // TODO this code should be adjusted/removed when a proper content-locale handling is implemented
```

Comment:
Should we maybe remove the comment? What better approach of defining a `contentLocale` at the very beginning do we have than the code below?

Reply:
I think we should use something which does not run `async` as currently this could end in a race condition when localizations are still loaded.

I think the best default value would be the first language which I have permission for, so the first available language of the first userRole.

Maybe we introduce a `getContentLocale` method on the User object itself as it could be interesting I think for PHP aswell or what do you think?

e.g.:

```diff
diff --git a/src/Sulu/Bundle/AdminBundle/Resources/js/stores/userStore/types.js b/src/Sulu/Bundle/AdminBundle/Resources/js/stores/userStore/types.js
index b2a3577bf0..0bff12df17 100644
--- a/src/Sulu/Bundle/AdminBundle/Resources/js/stores/userStore/types.js
+++ b/src/Sulu/Bundle/AdminBundle/Resources/js/stores/userStore/types.js
@@ -3,6 +3,7 @@
 export type User = {
     id: number,
     locale: string,
+    contentLocale: string,
     settings: {[string]: string},
     username: string,
 };
diff --git a/src/Sulu/Bundle/AdminBundle/Resources/js/stores/userStore/userStore.js b/src/Sulu/Bundle/AdminBundle/Resources/js/stores/userStore/userStore.js
index 0491739074..e7f1549484 100644
--- a/src/Sulu/Bundle/AdminBundle/Resources/js/stores/userStore/userStore.js
+++ b/src/Sulu/Bundle/AdminBundle/Resources/js/stores/userStore/userStore.js
@@ -61,20 +61,7 @@ class UserStore {
             this.persistentSettings.set(key, persistentSettings[key]);
         });
 
-        const contentLocale = this.persistentSettings.get(CONTENT_LOCALE_SETTING_KEY);
-
-        if (contentLocale) {
-            this.contentLocale = contentLocale;
-        }
-
-        if (!contentLocale) {
-            // load and use first (default) localization of first webspace as content-locale for the user
-            localizationStore.loadLocalizations().then(action((localizations) => {
-                const defaultLocalizations = localizations.filter((localization) => localization.default);
-                const fallbackLocalization = defaultLocalizations.length ? defaultLocalizations[0] : localizations[0];
-                this.contentLocale = fallbackLocalization ? fallbackLocalization.locale : this.contentLocale;
-            }));
-        }
+        this.contentLocale = user.contentLocale;
     }
 
     @action updateContentLocale(contentLocale: string) {
diff --git a/src/Sulu/Bundle/SecurityBundle/Entity/User.php b/src/Sulu/Bundle/SecurityBundle/Entity/User.php
index 9f3820e1bd..fa1ab3e956 100644
--- a/src/Sulu/Bundle/SecurityBundle/Entity/User.php
+++ b/src/Sulu/Bundle/SecurityBundle/Entity/User.php
@@ -32,6 +32,8 @@ use Symfony\Component\Security\Core\User\UserInterface as SymfonyUserInterface;
  */
 class User extends ApiEntity implements UserInterface, Serializable, EquatableInterface
 {
+    public const USER_SETTINGS_KEY_CONTENT_LOCALE = 'sulu_admin.content_locale';
+
     /**
      * @var int
      * @Expose
@@ -678,6 +680,29 @@ class User extends ApiEntity implements UserInterface, Serializable, EquatableIn
         return $userSettingValues;
     }
 
+    /**
+     * @VirtualProperty
+     * @Groups({"frontend"})
+     */
+    public function getContentLocale(): string
+    {
+        $contentLocale = $this->getSettings()[self::USER_SETTINGS_KEY_CONTENT_LOCALE] ?? null;
+
+        if ($contentLocale) {
+            return $contentLocale;
+        }
+
+        foreach ($this->userRoles as $userRole) {
+            $locales = $userRole->getLocales();
+
+            if (isset($locales[0])) {
+                return $locales[0];
+            }
+        }
+
+        return $this->getLocale();
+    }
+
     /**
      * Set contact.
      *
```

Reply:
I don't see how the current solution would end up in a race condition... The worst thing that can happen is that initially it loads without language, and then a second request happens, although I have never seen that happening.

However, I think it might still make sense to put this logic on the server, so that eventual other clients don't have to implement that as well. But I think we should get rid of this bug ASAP, and not discuss this now.